[tt-train] Update tt-train code owners: replace rfurko-tt with new team members

mdragulaTT opened the pull request:
Remove rfurko-tt and add mdragulaTT, vmelnykovTT, dyurshevichTT, and zbaczewskiTT as owners for ops and optimizers sections in tt-train.

### Ticket
N/A

### Problem description
Code ownership for tt-train ops and optimizers sections needed to be updated to reflect current team structure.

### What's changed
Updated CODEOWNERS file for tt-train repository sections:
- Added @mdragulaTT, @vmelnykovTT, @dyurshevichTT, @zbaczewskiTT tt-train/sources/ttml/metal/** 
- Added @mdragulaTT, @vmelnykovTT, @dyurshevichTT tt-train/sources/ttml/ops/**
- Added @zbaczewskiTT as owner for tt-train/sources/ttml/optimizers/**
- Applied same ownership structure to corresponding test directories

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

mdragulaTT committed:
Update tt-train code owners: replace rfurko-tt with new team members

Remove rfurko-tt and add mdragulaTT, vmelnykovTT, dyurshevichTT, and zbaczewskiTT as owners for ops and optimizers sections in tt-train.

tt-rkim commented on .github/CODEOWNERS:
```diff
@@ -346,7 +346,12 @@ docs/source/tt-metalium/tt_metal/apis/kernel_apis/data_movement/ @abhullar-tt @r
 
 # tt-train
-tt-train/** @dmakoviichuk-tt @rfurko-tt @ayerofieiev-tt @tenstorrent/codeowner-bypass
+tt-train/** @dmakoviichuk-tt @ayerofieiev-tt @tenstorrent/codeowner-bypass
```
Artem will co-own everything with Denys, that doesn't have a specific heading below?

dmakoviichuk-tt reviewed (approved):
Looks good but probably everyone from the team can own tests.

mdragulaTT committed:
Add whole tt-train team to tests ownership

Consolidate test ownership under single tt-train/tests/** entry with all team members (@mdragulaTT @vmelnykovTT @dyurshevichTT @zbaczewskiTT @athompsonTT @abustamanteTT) instead of separate per-subdirectory rules.
